packages/messageport/tizen/src/messageport.h

```diff
@@ -2,59 +2,127 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#ifndef MESSAGEPORT_H
-#define MESSAGEPORT_H
+#ifndef FLUTTER_PLUGIN_MESSAGEPORT_H_
+#define FLUTTER_PLUGIN_MESSAGEPORT_H_
 
-#include <flutter/event_channel.h>
-#include <flutter/standard_method_codec.h>
 #include <message_port.h>
 
+#include <functional>
 #include <map>
-#include <set>
+#include <memory>
+#include <optional>
+#include <string>
+#include <variant>
 
-typedef std::unique_ptr<flutter::EventSink<flutter::EncodableValue>> EventSink;
+class MessagePortError {
+ public:
+  explicit MessagePortError(int error_code)
+      : error_code_(error_code),
+        error_message_(get_error_message(error_code_)) {}
+
+  const std::string& message() const { return error_message_; }
+
+  int code() { return error_code_; }
+
+ private:
+  int error_code_;
+  std::string error_message_;
+};
+
+template <class T>
+class ErrorOr {
+ public:
+  ErrorOr(const T& other) { new (&vlaue_or_error_) T(other); }
 
-struct MessagePortResult {
-  MessagePortResult() : error_code(MESSAGE_PORT_ERROR_NONE){};
-  MessagePortResult(int code) : error_code(code) {}
+  ErrorOr(const T&& other) { vlaue_or_error_ = std::move(other); }
 
-  // Returns false on error
-  operator bool() const { return MESSAGE_PORT_ERROR_NONE == error_code; }
+  ErrorOr(const MessagePortError& other) {
+    new (&vlaue_or_error_) MessagePortError(other);
+  }
 
-  std::string message() { return get_error_message(error_code); }
+  ErrorOr(const MessagePortError&& other) {
+    vlaue_or_error_ = std::move(other);
+  }
+
+  bool has_error() const {
+    return std::holds_alternative<MessagePortError>(vlaue_or_error_);
+  }
+
+  const T& value() const { return std::get<T>(vlaue_or_error_); };
+
+  const MessagePortError& error() const {
+    return std::get<MessagePortError>(vlaue_or_error_);
+  };
+
+ private:
+  ErrorOr() = default;
+
+  std::variant<T, MessagePortError> vlaue_or_error_;
+};
 
-  int error_code;
+struct Message {
+  const std::string remote_app_id;
+  const std::string remote_port;
+  bool trusted_remote_port = false;
```

Comment:
You may use the already existing `RemotePort` class to represent this information.